### PR TITLE
import: better error code

### PR DESCRIPTION
**Pull request by rerowep**
* Adds error message and code for failed requests

### CODE REVIEW

File: rero_ils/modules/imports/api.py [hunk 1]
```diff
@@ -370,11 +370,28 @@ def search_records(self, what, relation, where='anywhere', max_results=0,
...
+                    f'url: {url_api}'
+                )
+                return []
+
         if max_results == 0:
             max_results = self.max_results
         if self.name == 'LOC' and relation == "all":
             relation = "="
         self.init_results()
+        url_api = '???'
```
Comment (jma) [inline rero_ils/modules/imports/api.py:394]: ???

File: rero_ils/modules/imports/api.py [hunk 2]
```diff
@@ -395,97 +413,90 @@ def search_records(self, what, relation, where='anywhere', max_results=0,
...
-            # abort(503, description='Timeout')
+            self.create_aggregations(self.results)
+        except requests.exceptions.ConnectionError as error:
+            self.status_code = 433
+            self.status_msg = str(error)
+        except requests.exceptions.HTTPError as error:
+            current_app.logger.error(f'{type(error)} {error}')
+            self.status_code = error.response.status_code
+            self.status_msg = str(error)
+            import traceback
```
Comment (jma) [inline rero_ils/modules/imports/api.py:471]: Please put this on the top if you can.

File: rero_ils/modules/imports/api.py [hunk 3]
```diff
@@ -395,97 +413,90 @@ def search_records(self, what, relation, where='anywhere', max_results=0,
...
+            }
         return self.results, self.status_code
 
 
 class BnfImport(Import):
     """Import class for BNF."""
 
     name = 'BNF'
-    url = 'http://catalogue.bnf.fr'
+    url = 'https://catalogue.bnf.fr'
```
Comment (jma) [inline rero_ils/modules/imports/api.py:499]: Is this should be moved in the configuration one day?

File: tests/api/test_external_services.py [hunk 4]
```diff
@@ -858,3 +859,54 @@ def test_documents_import_kul_isbn(mock_get, client, kul_anywhere_123,
     ))
     assert res.status_code == 200
     assert get_json(res).get('metadata', {}).get('ui_title_text')
+
+
+@mock.patch('requests.get')
+@mock.patch('rero_ils.permissions.login_and_librarian', mock.MagicMock())
+def test_documents_import_bnf_errors(mock_get, client):
```
Comment (jma) [inline tests/api/test_external_services.py:866]: great!

**Review by jma — APPROVED** [review]

File: rero_ils/modules/imports/api.py [hunk 5]
```diff
@@ -395,97 +413,90 @@ def search_records(self, what, relation, where='anywhere', max_results=0,
...
-                            'code': self.status_code,
-                            'title': f'{self.name}: Bad status code!',
-                            'detail': f'Status code: {response.status_code}'
+                self.status_code = response.status_code
+                self.status_msg = 'Request error.'
+                response.raise_for_status()
+
+                xml_records = _split_stream(BytesIO(response.content))
+
+                for xml_record in xml_records:
```
Comment (zannkukai) [inline rero_ils/modules/imports/api.py:422]: could be merged as "xml_records" in never used anywhere else.
```for xml_record in _split_stream(BytesIO(response.content)):```

File: rero_ils/modules/imports/api.py [hunk 6]
```diff
@@ -395,97 +413,90 @@ def search_records(self, what, relation, where='anywhere', max_results=0,
...
+                    # convert xml in marc json
+                    json_data = create_record(xml_record)
+                    json_data = self.clean_marc(json_data)
```
Comment (zannkukai) [inline rero_ils/modules/imports/api.py:425]: could be merged too
```json_data = self.clean_marc(create_record(xml_record))```

**Review by zannkukai — APPROVED** [review]
only cosmetic messages

File: rero_ils/modules/imports/api.py (hunk 3 above)
Comment (rerowep) [reply rero_ils/modules/imports/api.py:499]: Maybe, but this is already an class for BNF import. normally this URL should not change.

**Review by PascalRepond — APPROVED** [review]

**Pull request merged**